aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -368,9 +364,18 @@ Tensor empty_like(
 Tensor new_empty(
     const Tensor& self,
     IntArrayRef size,
-    const TensorOptions& options
+    c10::optional<ScalarType> dtype_opt,
+    c10::optional<Layout> layout_opt,
+    c10::optional<Device> device_opt,
+    c10::optional<bool> pin_memory_opt
     ) {
-  return at::empty(size, self.options().merge_in(options));
+  return at::empty(size,
+    dtype_opt.value_or_else([&] {return typeMetaToScalarType(self.options().dtype());}),
+    layout_opt.value_or(self.options().layout()),
+    device_opt.value_or(self.options().device()),
+    pin_memory_opt.value_or(self.options().pinned_memory()),
```

Comment:
This is misleading because there is never actually pinned memory set on `self.options`

```
TensorOptions Tensor::options() const {
  return TensorOptions().dtype(dtype())
                        .device(device())
                        .layout(layout());
}
```